tools: add DM benchmark report (#895)
pingcap/docs: tools/dm/benchmark.md

> ---

Reviewer: Another thing: since we already have a folder named "benchmark" in the root directory, please **move this file to docs/benchmark and rename this file to dm-benchmark.md**. Also have checked with @GregoryIan.

Author: done, I have moved this file to `benchmark/dm-v1-alpha.md` which follows the name regulation under this folder

title: DM Benchmark Report

> summary: DM benchmark report

Reviewer: ```suggestion
summary: Learn about the DM benchmark report.
```

category: tools
---

# DM Benchmark Report

> 

Reviewer: Adding a brief summary sentence under the 1st level heading would be better:) 

"This DM benchmark report describes the test purpose, environment, scenario, and result."

## Test purpose

> The purpose of this test is to test the performance of DM incremental synchronization

Reviewer: ```suggestion
The purpose of this test is to test the performance of DM incremental replication.
```

| 192.168.0.9  | CentOS Linux release 7.6.1810 | 3.10.0-957.1.3.el7.x86\_64   | ext4             |
| 192.168.0.10 | CentOS Linux release 7.6.1810 | 3.10.0-957.1.3.el7.x86\_64   | ext4             |
| 192.168.0.11 | CentOS Linux release 7.6.1810 | 3.10.0-957.1.3.el7.x86\_64   | ext4             |

Hardware information:

| Type         | 192.168.0.9, 192.168.0.10, 192.168.0.11               |  192.168.0.6, 192.168.0.7, 192.168.0.8                |
|--------------|-------------------------------------------------------|-------------------------------------------------------|
| CPU          | 8 vCPUs, Intel(R) Xeon(R) Platinum 8163 CPU @ 2.50GHz | 4 vCPUs, Intel(R) Xeon(R) Platinum 8163 CPU @ 2.50GHz |
| Memory       | 16G                                                   | 8G                                                    |
| Disk         | 1T Aliyun ESSD                                        | 256G Aliyun ESSD                                      |
| Network card | 1 Gigabit Ethernet, 1000Mb/s                          | 1 Gigabit Ethernet, 1000Mb/s                          |

### Cluster topology

> | Machine IP   | Deployment Instance |

Reviewer: ```suggestion
| Machine IP   | Deployment instance |
```

> | Machine IP   | Deployment Instance |

Reviewer: ```suggestion
| Machine IP   | Deployment instance |
```

Reviewer: Keep the format consistent :)

|--------------|------------|
| 192.168.0.9  | TiKV \* 1, TiDB \* 1  |
| 192.168.0.10 | TiKV \* 1  |
| 192.168.0.11 | TiKV \* 1  |
| 192.168.0.6  | PD \* 1, MySQL \* 1, DM-worker \* 1  |
| 192.168.0.8  | PD \* 1, MySQL \* 1, DM-worker \* 1  |
| 192.168.0.7  | PD \* 1, DM-master \* 1  |

### Version information

- MySQL version: 5.7.25-log

- TiDB version: v3.0.0-beta-27-g6398788

- DM version: v1.0.0-alpha-10-g4d01d79

- Sysbench version: 1.0.9

> ## Test Scenario

Reviewer: ```suggestion
## Test scenario
```

### Data flow

MySQL1 (192.168.0.8) -> DM-worker1 (192.168.0.6) -> TiDB (192.168.0.9)

### Test procedure

> - setup environment

Reviewer: ```suggestion
- Set up environment
```

> - use sysbench to create talbe and generate initial data in upstream MySQL

Reviewer: ```suggestion
- Use sysbench to create the table and generate the initial data in upstream MySQL
```

> - start DM-task in `all` mode

Reviewer: ```suggestion
- Start DM-task in the `all` mode
```

> - use sysbench to generate incremental data in upstream MySQL

Reviewer: ```suggestion
- Use sysbench to generate incremental data in upstream MySQL
```

### Use sysbench to generate data load in upstream MySQL

Upstream sysbench test script:

```
sysbench --test=oltp_insert --tables=2 --num-threads=1024 --mysql-host=192.168.0.8 --mysql-port=3306 --mysql-user=root --mysql-db=dm_poc --db-driver=mysql --report-interval=10 --time=900 run
```

The structure of the table used for the test:

``` sql
CREATE TABLE `sbtest` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `k` int(11) NOT NULL DEFAULT '0',
  `c` char(120) CHARSET utf8mb4 COLLATE utf8mb4_bin NOT NULL DEFAULT '',
  `pad` char(60) CHARSET utf8mb4 COLLATE utf8mb4_bin NOT NULL DEFAULT '',
  PRIMARY KEY (`id`),
  KEY `k_1` (`k`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
```

> ### The deployment and configuration details:

Reviewer: ```suggestion
### The deployment and configuration details
```

```
// TiKV configuration
sync-log = false
[defaultcf]
block-cache-size = "4GB"
[writecf]
block-cache-size = "4GB"
[raftdb.defaultcf]
block-cache-size = "4GB"

// DM task syncer unit configuration
syncer:
    worker-count: 256
    batch: 100
    max-retry: 20
```

## Test result

| items | threads | qps                                       | tps    | 95% Latency (ms)            |
| ----- | ------- | ----------------------------------------- | ------ | --------------------------- |
| MySQL | 1024    | 15.10k                                    | 15.10k | 121.08                      |
| DM    | 256     | 13.89k                                    | 13.89k | 210 (txn execution latency) |
| TiDB  | -       | 18.53k (Begin/Commit 2.4k Replace 13.80k) | 2.27k  | 29                          |

> ### Monitor screenshot

Reviewer: ```suggestion
### Monitor screenshots
```

> ### DM key indicator monitor

Reviewer: ```suggestion
#### DM key indicator monitor
```

![](../../media/dm-benchmark-01.png)

> ### TiDB key indicator monitor

Reviewer: ```suggestion
#### TiDB key indicator monitor
```

---
title: DM Benchmark Report
summary: DM benchmark report

> category: benchmark

Reviewer 2: ```suggestion
category: tools
```
?

---

# DM Benchmark Report

## Test purpose

> The purpose of this test is to test the performance for DM incremental synchronization

Reviewer 3: `for`？ or `of`

| 192.168.0.10 | CentOS Linux release 7.6.1810 | 3.10.0-957.1.3.el7.x86\_64   | ext4             |
| 192.168.0.11 | CentOS Linux release 7.6.1810 | 3.10.0-957.1.3.el7.x86\_64   | ext4             |

Hardware information:

| Type         | 192.168.0.9, 192.168.0.10, 192.168.0.11               |  192.168.0.6, 192.168.0.7, 192.168.0.8                |
|--------------|-------------------------------------------------------|-------------------------------------------------------|
| CPU          | 8 vCPUs, Intel(R) Xeon(R) Platinum 8163 CPU @ 2.50GHz | 4 vCPUs, Intel(R) Xeon(R) Platinum 8163 CPU @ 2.50GHz |
| Memory       | 16G                                                   | 8G                                                    |
| Disk         | 1T Aliyun ESSD                                        | 256G Aliyun ESSD                                      |
| Network card | 1 Gigabit Ethernet, 1000Mb/s                          | 1 Gigabit Ethernet, 1000Mb/s                          |

### Cluster topology

| Machine IP   | Deployment Instance |
|--------------|------------|
| 192.168.0.9  | TiDB \* 1  |

> | 192.168.0.9  | TiKV \* 1  |

Reviewer 2: same IP only keep one line? like `TiDB * 1; TiKV * 1`

| 192.168.0.10 | TiKV \* 1  |
| 192.168.0.11 | TiKV \* 1  |
| 192.168.0.6  | PD \* 1    |
| 192.168.0.7  | PD \* 1    |
| 192.168.0.8  | PD \* 1    |
| 192.168.0.6  | MySQL \* 1 |
| 192.168.0.8  | MySQL \* 1 |
| 192.168.0.6  | DM-worker \* 1 |
| 192.168.0.8  | DM-worker \* 1 |
| 192.168.0.8  | DM-worker \* 1 |
| 192.168.0.7  | DM-master \* 1 |

> ### version information

Reviewer 2: ```suggestion
### Version information
```
